baselines/FedSMOO/FedSMOO/main.py

```diff
@@ -0,0 +1,176 @@
+
+    # testloader = None
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    evaluate_fn = gen_evaluate_fn(testloader, device, cfg.model) 
+    
+    strategy = instantiate(cfg.strategy,
+                           evaluate_fn=evaluate_fn)
+
+    
+    #-----------------------------------------------------------------------------------------------
+    # 4.5 Create a custom server?
+
+    client_manager = SimpleClientManager()
+    
+    if cfg.method == "FedSMOO": 
+        server = instantiate(cfg.server,
+                        net = instantiate(cfg.model).to(device),
+                        sam_lr = cfg.sam_lr,
+                        client_manager = client_manager,
+                        strategy = strategy)
+    
+    elif cfg.method == "FedAvg":
+        server = Server(client_manager = client_manager,
+                        strategy = strategy)
+    
+    else:
+        # use strategy to initialize the default server
+        print("write a suitable message")
+        server = None
+        
+    
+    #-----------------------------------------------------------------------------------------------
+    # 5. Start Simulation
+    
+    print("\n---------------------- starting simulation --------------------------\n")
+
+    history = fl.simulation.start_simulation(
+                    client_fn=client_fn,
+                    num_clients=cfg.num_clients,
+                    server=server,
+                    config=fl.server.ServerConfig(num_rounds=cfg.num_rounds),
+                    client_resources={"num_cpus": cfg.client_resources.num_cpus,
+                                    "num_gpus": cfg.client_resources.num_gpus,},
+                    strategy=strategy)
+    #------------------------------------------------------------------------------------------------
+
+    # 6. Save your results
+    # Here you can save the `history` returned by the simulation and include
+    # also other buffers, statistics, info needed to be saved in order to later
+    # on generate the plots you provide in the README.md. You can for instance
+    # access elements that belong to the strategy for example:
+    # data = strategy.get_my_custom_data() -- assuming you have such method defined.
+    # Hydra will generate for you a directory each time you run the code. You
+    # can retrieve the path to that directory with this:
+    # save_path = HydraConfig.get().runtime.output_dir
+
+    accuracy_test = history.metrics_centralized["accuracy"]
+    plot_fn(cfg, accuracy_test)
+
+    if cfg.dataset.dirichlet:
+        dataset_prep = "dirichlet"
+    else:
+        dataset_prep = "pathological"
+
+    logfile = f"./results/{cfg.method}_{cfg.dataset.dataset_name}_{cfg.num_clients}_{dataset_prep}.log"
+    logging.basicConfig(filename=logfile, 
+					format='%(asctime)s %(message)s', 
+					filemode='w') 
+    logger=logging.getLogger() 
+    logger.setLevel(logging.DEBUG) 
+    logger.info(accuracy_test) 
```

Comment:
Hydra creates a log for you. Although the current integration of hydra+flower is not 100% great, do you need the above?